app/waf/src/repositories/common.ts

```diff
@@ -22,4 +31,42 @@ export class CommonRepository<
 
     return super.updateById(id, data, options);
   }
+
+  findEntitiesByTenant(tenantId: string): Promise<T[]> {
+    const filter: Filter = {where: {tenantId: tenantId}};
+    // maybe remove this "magic conversion" someday.
+    return this.find(filter as (Filter<T>));
+  }
+
+  async findEntityByTenant(tenantId: string, id: ID): Promise<T> {
+    // have to do this, since findById filter does not work at all.
+    const filter: Filter = {where: {and: [{tenantId: tenantId}, {id: id}]}};
+
+    let data = await this.find(filter as (Filter<T>));
+    if (data.length !== 0) {
+      this.logger.debug('find resource ' + data[0]);
+      // return the first one we found.
+      return data[0];
+    } else {
+      throw new EntityNotFoundError(this.entityClass.name, id);
+    }
+  }
+
+  async updateEntityByTenant(
+    tenantId: string,
+    id: ID,
+    entitry: T,
+  ): Promise<void> {
+    const data = await this.findEntityByTenant(tenantId, id);
+
+    this.logger.debug('update resource ' + data + ', with ' + entitry);
+    return super.updateById(id, entitry);
+  }
+
+  async deleteEntityByTenant(tenantId: string, id: ID): Promise<void> {
```

Comment:
Why do we need to find it by both tenant and id before delete? Cannot ID distinguish an entity?

Reply:
I think it's better to do a tenant id check implicitly before we do any to that entity.